unstructured/ingest/connector/salesforce.py

```diff
@@ -87,6 +87,16 @@ class SalesforceIngestDoc(IngestDocCleanupMixin, BaseIngestDoc):
     record_type: str
     record_id: str
     registry_name: str = "salesforce"
+    _record: OrderedDict = field(default_factory=lambda: OrderedDict())
+
+    def __post_init__(self):
+        self.update_source_metadata()
```

Comment:
why calling this in post_init rather than opportunistically? seems like this defeats the point of lazy fetching right? also this guarantees were doing this both when the connector creates the ingest_doc and when we rehydrate in processing, right?

Reply:
I wanted to move it out of the `get_file` method since that can be skipped if the file exists and then all this metadata wouldn't exist. This was the simplest approach but I see you point. I can further refactor to get the data.